src/mcts/search.cc

```diff
@@ -710,6 +715,21 @@ void SearchWorker::GatherMinibatch() {
   int minibatch_size = 0;
   int collision_events_left = params_.GetMaxCollisionEvents();
   int collisions_left = params_.GetMaxCollisionVisitsId();
+  if (params_.GetMaxConcurrentSearchers() != 0) {
+    while (true) {
+      if (search_->stop_.load(std::memory_order_acquire)) return;
+      int available =
+          search_->pending_searchers_.load(std::memory_order_acquire);
+      if (available > 0 &&
+          search_->pending_searchers_.compare_exchange_weak(
+              available, available - 1, std::memory_order_acq_rel)) {
+        break;
+      }
+      // This is a hard spin lock to reduce latency but at the expense of busy
+      // wait cpu usage. If search worker count is large, this is probably a bad
+      // idea.
+    }
+  }
```

Comment:
As in multithread situation it can happen that many threads do a busy wait in that place, maybe it makes sense to add `if (!search_->IsSearchActive()) return;` after it?

Otherwise after finishing the search, every thread will do one full evaluation loop with NN request.

Reply:
Note that IsSearchActive is the same as the first if check in this loop.